### PR TITLE
Rework tests

### LINKED ISSUE #9: More extensible tests?
### 🤔 What's the problem you've observed?

We're adding tests in the same file.

### ✨ Do you have a proposal for making it better?

Could draw inspiration from https://github.com/tooky/retro-tools and loop over separate test scripts in a directory.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/developer_experience.md)*

### PR DESCRIPTION
**Pull request by mattwynne**
### 🤔 What's changed?

Reworked tests to:
* use a temporary git repo each time (instead of running in the project's own git repo)
* run in a loop (fixes #9)
* allow running a single test or all of them
* inject release date (so we can simplify the tests)
* set the tag in the tests's working directory git repo, so we no longer need the `--no-git-tag-check` option
* allow setting expectations with `expected.git-log` as well as `expected.git-diff` or `expected.output` files

### ⚡️ What's your motivation? 

I want to be able to TDD changes, and this gives more options for testing.

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: tests/release-and-post-release-with-commit.sh.expected.git-log [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Prepare for the next development iteration  (HEAD -> main)
+Prepare release v1.0.0  (tag: v)
```
Comment (mattwynne) [inline tests/release-and-post-release-with-commit.sh.expected.git-log:2]: Note that the expectation here is wrong, it's just characterizing the current buggy behaviour. This line should read:

    Prepare release v1.0.0  (tag: v1.0.0)

File: polyglot-release [hunk 2]
```diff
@@ -201,7 +197,10 @@ release_javascript
 release_java
 release_ruby
 
-changelog release "$NEW_VERSION" --tag-format "v%s" -o CHANGELOG.md
+changelog release "$NEW_VERSION" \
+  --release-date $RELEASE_DATE \
```
Comment (mpkorstanje) [inline polyglot-release:201]: Does this required installing a new version of `changelog`?

Comment (mpkorstanje) [reply polyglot-release:201]: Also looks like `RELEASE_DATE` is not initialized.

File: polyglot-release [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 set -e
 
+RELEASE_DATE=${RELEASE_DATE:-$(date +%F)}
```
Comment (mpkorstanje) [inline polyglot-release:4]: Should be initialized in the `# Initialize global variables` section.

Comment (mpkorstanje) [reply polyglot-release:4]: I don't like that `RELEASE_DATE` is now a secret input. This should be part of the public interface.

File: polyglot-release-test [hunk 4]
```diff
@@ -1,53 +1,57 @@
 #!/bin/bash
 
 FAILURE=
-
-function start_test() {
-    echo -n $1
+SRC=$(realpath .)
+
+function setup_git_repo() {
+  git init --quiet --initial-branch=main
```
Comment (mpkorstanje) [inline polyglot-release-test:7]: Removed the `--initial-branch=main`. Doesn't work on my version of git.

**Review by mpkorstanje — APPROVED** [review]

**Comment by mattwynne** [comment]
🎉

**Comment by mattwynne** [comment]
Thanks Rien!

File: polyglot-release (hunk 3 above)
Comment (mattwynne) [reply polyglot-release]: Meh. I think unless we can see a need for a user to do it, there's no need to clutter up the UI.

**Pull request merged**